src/jira/api.rs

```diff
@@ -337,4 +363,25 @@ mod tests {
         let versions = parse_pending_versions(&search, "the-field");
         assert_eq!(expected, versions);
     }
+
+    #[test]
+    fn test_comment_issue() {
+        let jira_config = JiraConfig {
+            host: "the-host".into(),
+            username: "the-jira-user".into(),
+            password: "the-jira-pass".into(),
+            progress_states: Some(vec!["progress1".into()]),
+            review_states: Some(vec!["reviewing1".into()]),
+            resolved_states: Some(vec!["resolved1".into(), "resolved2".into()]),
+            fixed_resolutions: Some(vec!["it-is-fixed".into()]),
+            fix_versions_field: Some("the-versions".into()),
+            pending_versions_field: Some("the-pending-versions".into()),
+            restrict_comment_visibility_to_role: None,
+        };
+        let jira_sess = match JiraSession::new(&jira_config) {
+            Ok(j) => j,
+            Err(e) => panic!(e.to_string()),
+        };
+        // jira_sess.comment_issue("oh", "yeah");
```

Comment:
assuming the testing is part of the "WIP" status here?

Reply:
looking back at the jira tests, i think you want to put this into `jira_workflow_test` maybe and use `mock_jira`

Reply:
I looked at that, but I'm trying to test a method within the JiraSession impl (comment_issue) and looks like `jira_workflow_test` mocks the whole Jira object and its APIs. It seems like I need a `jira_api_test` or something (which is why I was doing the test inline in the file), but seems pretty involved as it looks like I'd have to mock the HTTPClient or dependency inject one into the `new` function.